Content.Server/Fluids/EntitySystems/SpraySystem.cs

```diff
@@ -57,42 +57,40 @@ private void OnAfterInteract(EntityUid uid, SprayComponent component, AfterInter
             return;
         }
 
-        var playerPos = Transform(args.User).Coordinates;
+        var playerMapPos = Transform(args.User).MapPosition;
+        var clickMapPos = args.ClickLocation.ToMap(EntityManager);
 
-        if (args.ClickLocation.GetGridUid(EntityManager) != playerPos.GetGridUid(EntityManager))
-            return;
-
-        var direction = (args.ClickLocation.Position - playerPos.Position).Normalized;
-        var threeQuarters = direction * 0.75f;
-        var quarter = direction * 0.25f;
+        var mapDirection = (clickMapPos.Position - playerMapPos.Position).Normalized;
```

Comment:
This will throw if their positions are the same so should probably check if the length is 0.